src/MatchingResult.php

```diff
@@ -61,7 +57,7 @@ public function methods(): array
 
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        if ($this->success === false) {
+        if ($this->success === false || $this->route === null) {
```

Comment:
There's no need for this check. It will be non-null if success is `true`.

Reply:
It's true for now, but what will be after a year of changes?

Reply:
We'll see in a year. What we need to think upfront are interfaces, not implementation detail. It should solve a problem and just that.